pkg/provider/azure_vmss.go

```diff
@@ -1682,24 +1682,76 @@ func getScaleSetAndResourceGroupNameByIPConfigurationID(ipConfigurationID string
 }
 
 func (ss *ScaleSet) ensureBackendPoolDeletedFromVMSS(backendPoolID, vmSetName string) error {
-	vmssNamesMap := make(map[string]bool)
+	if !ss.useStandardLoadBalancer() || ss.EnableMultipleStandardLoadBalancers {
+		found := false
 
-	// the standard load balancer supports multiple vmss in its backend while the basic sku doesn't
-	if ss.useStandardLoadBalancer() {
 		cachedUniform, err := ss.vmssCache.Get(consts.VMSSKey, azcache.CacheReadTypeDefault)
 		if err != nil {
 			klog.Errorf("ensureBackendPoolDeletedFromVMSS: failed to get vmss uniform from cache: %v", err)
 			return err
 		}
+		vmssUniformMap := cachedUniform.(*sync.Map)
+
+		vmssUniformMap.Range(func(key, value interface{}) bool {
+			vmssEntry := value.(*VMSSEntry)
+			if to.String(vmssEntry.VMSS.Name) == vmSetName {
```

Comment:
Why do we need to check the vmSetName here?

Reply:
Because at this point, we don't know if this vmSetName is Uniform or Flex yet

Reply:
Why does the name matter? Do you mean there could be vmss flex in vmssUniformMap?

Reply:
When the if logic on line 1685 is true, CCM only needs to delete backend Pool from the VMSS of vmSetName. However, CCM does not know if vmSetName is a Flex scale set or Uniform scale set. So, it first goes through the Unifrom VMSS cache and check if any VMSS in the cache has name "vmSetName". If it does not find any, then it continues to go through the VMSS Flex cache.

Reply:
The line 1685 means it is using basic LB or using multiple SLB

Reply:
It seems we don't need to check if we use multi-slb or not. We just claim here that we want vmss A, B and C to decouple with the LB. If there is no LB ref on A and B, it will be two no-ops.

Reply:
Make sense. I removed checking multi-slb just now.